src/Location.cpp

```diff
@@ -0,0 +1,61 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   Location.cpp                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lcouto <lcouto@student.42sp.org.br>        +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/04/21 21:53:47 by lcouto            #+#    #+#             */
+/*   Updated: 2023/04/21 21:53:47 by lcouto           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+# include "Location.hpp"
+
+Location::Location(void)
+{
+	return ;
+}
+
+Location::~Location(void) 
+{
+	return ;
+}
+
+void    Location::insertLocationData(std::pair<std::string, std::vector<std::string> > directive)
+{
+    bool    isUniqueDirective = (directive.first == "root" || directive.first == "autoindex");
+    bool    directiveNotOnMap = (this->_locationData.find(directive.first) != this->_locationData.end());
+
+    if (isUniqueDirective)
+        if (directiveNotOnMap)
+            throw (DuplicateDirectiveError());
+        else
+            this->_locationData.insert(directive);
+    else
+        if (directiveNotOnMap)
+            this->_locationData.insert(directive);
+        else
+            this->_locationData[directive.first].insert(this->_locationData[directive.first].begin(), directive.second.begin(), directive.second.end());

```

Comment:
ta codando em python po KKKKKKKKKKK
acho interessante adicionar as chaves nem que seja só nas condições mais externas

```suggestion

    if (isUniqueDirective) {
        if (directiveNotOnMap)
            throw (DuplicateDirectiveError());
        else
            this->_locationData.insert(directive);
    } else {
        if (directiveNotOnMap)
            this->_locationData.insert(directive);
        else
            this->_locationData[directive.first].insert(this->_locationData[directive.first].begin(), directive.second.begin(), directive.second.end());
    }
```